src/zenml/zen_server/routers/users_endpoints.py

```diff
@@ -271,6 +284,12 @@ def activate_user(
         """
         user = zen_store().get_user(user_name_or_id)
 
+        if user.is_admin != user_update.is_admin:
+            raise HTTPException(
+                status_code=status.HTTP_403_FORBIDDEN,
+                detail="Cannot update admin status of a user during activation.",
+            )
```

Comment:
```suggestion
            raise AuthorizationException(
                "Cannot update admin status of a user during activation.",
            )
```

We usually avoid raising `HTTPException` in the endpoints explicitly, but use regular exceptions that the `@handle_exceptions` decorator is able to convert into `HTTPException` automatically. This allows more specific, code-friendly exceptions to be unpacked on the other side (client/CLI).

Reply:
I recommend you just silently remove the `is_admin` field from the update. This can only be exploited through the REST API anyway.